rxandroidble/src/main/java/com/polidea/rxandroidble/internal/util/IllegalOperationChecker.java

```diff
@@ -79,18 +83,30 @@ public Object call() throws Exception {
                             propertiesIntToString(neededProperties, possibleProperties),
                             neededProperties
                     );
-                    handleMessage(message);
+                    MismatchData mismatchData = new MismatchData(message,
```

Comment:
I think that having a `MismatchData` here is superfluous and it only adds an allocation. The data could be passed directly to the `MismatchDataHandler`